### DISCUSSION TITLE
How can i remove this icon?

### DISCUSSION THREAD

**Discussion by EwertonBatista** [rainbow-me/rainbowkit#392] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/40125335/168899372-07a364ea-766a-4a59-8214-01f55489aee0.png)

**Comment by nickbytes** [comment]
@EwertonBatista there's currently no way of removing this icon, unless your users have an ENS profile avatar – in which case, that image will show there.

We've been talking through this internally this week and have been trying to decide whether we believe it is a little too opinionated to add this emoji image to your UI. Generally, we feel like we may have been a little too opinionated. We've thought of a few solutions:

1. allow some sort of prop for hiding the emoji, and only using a color there (similar to OpenSea, Zora, etc), like `showEmoji`
2. allow full customization of user avatars by passing in a component (along w/ the account prop)
3. allow users to remove the avatar altogether
4. show better examples for using a custom button + triggering a completely custom modal 

What would you like to see in place of this icon here? Or what option makes the most sense to you?

**Reply by quagliero** [reply]
Option 2 makes the most sense imo (so we can use stuff like blockies for those familiar with their metamask account icons), or failing that option 1 where it's togglable and just shows a plain colour.

**Reply by EwertonBatista** [reply]
I think option 2 is a good choice, but I think it would also be necessary to have an option to hide the icon, like this 
![image](https://user-images.githubusercontent.com/40125335/169165483-b3339402-45e7-4505-ba02-8bb6da2cebf9.png)

**Reply by nickbytes** [reply]
@EwertonBatista the spacing at the top looks a little empty there IMHO, but I'll talk w/ our designers and see what they suggest.

In the meantime, if you'd like to get something that looks like this, I'd suggest creating a [custom button](https://www.rainbowkit.com/docs/custom-connect-button), then replacing your `openAccountModal` button (from the example) w/ a button that opens your own custom modal.

**Reply by sicktastic** [reply]
I vote for option 2 as well, if not 3.

**Comment by carloshc** [comment]
Maybe four options: with this type of icon, the two MetaMask's types, and no icon at all.

![image](https://user-images.githubusercontent.com/1533025/169278130-2c8cb68e-bf1b-4ce1-98f1-7a8ec4a1d808.png)

**Reply by apecollector** [reply]
I implemented the jazzicons style icon:

```ts
import Jazzicon, { jsNumberForAddress } from "react-jazzicon";

const CustomAvatar: AvatarComponent = ({ address, ensImage, size }) => {
  return ensImage ? (
    <img src={ensImage} width={size} height={size} style={{ borderRadius: size }} />
  ) : (
    <Jazzicon diameter={size} seed={jsNumberForAddress(address)} />
  );
};
```

Then use it in the RainbowKitProvider:

```tsx
<RainbowKitProvider avatar={CustomAvatar}>
```

**Comment by nickbytes** [comment, marked as answer]
There's now a way to customize avatars, see our docs here: https://www.rainbowkit.com/docs/custom-avatars

**Reply by octavioamu** [reply]
thanks Nick, something Im trying to figure out is how to use that custom avatar with my custom connect button

**Reply by apecollector** [reply]
> thanks Nick, something Im trying to figure out is how to use that custom avatar with my custom connect button

https://github.com/rainbow-me/rainbowkit/discussions/392#discussioncomment-3802079
